core/src/main/java/com/linecorp/armeria/server/annotation/decorator/LoggingDecorator.java

```diff
@@ -59,6 +59,13 @@
      */
     float samplingRate() default 1.0f;
 
+    /**
+     * The rate at which to sample failed requests to log. Any number between {@code 0.0} and {@code 1.0} will
+     * cause a random sample of the requests to be logged. The random sampling is appropriate for low-traffic
+     * (ex servers that each receive &lt;100K requests). If unset, all requests will be logged.
+     */
+    float failedSamplingRate() default 1.0f;
```

Comment:
- Should we also need to add `successSamplingRate`?
- `failedSamplingRate` -> `failureSamplingRate` for consistency?
- `samplingRate()` and `failedSamplingRate()` are mutually exclusive. How about setting the default value of `failedSamplingRate` and `successSamplingRate` to `-1.0f`?
  We can override the `sampleRate` if they are positive.
```java
// In LoggingDecoratorFactoryFunction
float successSamplingRate = parameter.successSamplingRate() != -1.0f ? parameter.successSamplingRate() : parameter.samplingRate();
float failureSamplingRate = ...;
```